### PR TITLE
Edit User Route and Testing

### PR DESCRIPTION
**Pull request by jroodman**
This PR creates the edit and update user routes for HackerNews. Integration tests for the edit user endpoint are also included.

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -14,6 +15,29 @@ def new
...
+      @form_empty = "All form fields cannot be empty."
+      render :edit
+      return
+    end
+
+    if @user.update_attributes(parameters)
+      redirect_to root_path, notice: 'User successfully edited.'
+    else
+      render :edit
+    end
```
Comment (rysospicy) [inline app/controllers/users_controller.rb]: You could work the `parameters.empty?` logic into your main if/else statement:
```ruby
if parameters.any? && @user.update_attributes(parameters)
  redirect_to root_path, notice: 'User successfully edited.'
else
  @form_empty = "All form fields cannot be empty." if parameters.empty?
  render :edit
end
```

File: spec/features/user_editing_spec.rb [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+      fill_in 'Email', with: 'email@new.com'
+      fill_in 'Password', with: 'passwordnew123'
+      fill_in 'Password confirmation', with: 'passwordnew123'
+      click_button 'Submit'
+      expect(page).to have_content "User successfully edited"
+    end
+
+  end
+
+end
```
Comment (rysospicy) [inline spec/features/user_editing_spec.rb:110]: Real nice job with these tests!  👌

**Review by rysospicy — APPROVED** [review]
Good stuff.  👍

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -14,6 +15,29 @@ def new
     @user = User.new
   end
 
+  def update
+    # @user = User.find(session[:user_id])
+    @user = User.find(params[:id])
```
Comment (dce) [inline app/controllers/users_controller.rb:20]: Seems like this might let you update a user other than yourself? Is that the desired behavior?

Comment (jroodman) [reply app/controllers/users_controller.rb:20]: Will be changing that in the next pull request when login/logout is implemented. After discussing with Ryan, we decided this was the best choice for testing purposes until then.

**Comment by dce** [comment]
:+1: Looks good. Not sure about all this `@form_empty` stuff -- might be something to look to remove later on.

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (jroodman) [reply app/controllers/users_controller.rb]: Indeed! Thanks.

**Comment by jroodman** [comment]
@dce 
Sure, I'll go ahead and merge for now. Decided to use @form_empty to be able to give the user a different message for that error case. I considered using flash as a different option, but decided along with Ryan that flash was better suited to inter-page messages. Modifying the current use of flash to work to support this use as well would also likely convolute the code. 

Any recommendations about how to avoid @form_empty? I'm not in love with the use either and would be very interested in any of your thoughts.

**Comment by dce** [comment]
Just like, to get the form into an empty state, you'd have to go in and blank out your username and email, right? Seems like just saying "email can't be blank" is good enough.

**Pull request merged**